packages/flame_studio/lib/src/widgets/toolbar/flame_studio_toolbar.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flame_studio/src/core/theme.dart';
+import 'package:flame_studio/src/widgets/toolbar/pause_button.dart';
+import 'package:flame_studio/src/widgets/toolbar/start_button.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final toolbarHeightProvider = Provider((ref) => 28.0);
+
+class FlameStudioToolbar extends ConsumerWidget {
+  const FlameStudioToolbar({super.key});
+
+  /// List of widgets to show on the left side of the toolbar.
+  static final List<Widget> left = [];
+
+  /// List of widgets to show in the center of the toolbar.
+  static final List<Widget> middle = [
+    const StartButton(),
+    const PauseButton(),
+  ];
+
+  /// List of widgets to show on the right side of the toolbar.
+  static final List<Widget> right = [];
```

Comment:
I think it is probably better to create widgets from these lists instead of having static lists here

Reply:
The idea here is that it should be possible for an external code to insert new functionality into Flame Studio. In this case, you could say `FlameStudioToolbar.right.add(SpydonButton())`, and that button would appear inside the toolbar.

We may figure out a better way of handling this; maybe these lists will become riverpod providers as well so that the UI can update on the fly -- but for now we can keep them as simple lists.

Reply:
Hmm, this is recommended to do in Flutter, because the widget won't know that it should rebuild when these lists are modified, so depending on where the user edits these lists it might or might not work. If they should only be set from the beginning they could be passed in as arguments, or if it should be possible to change them during runtime they should indeed be providers.

Reply:
Right now it's impossible to say what the future requirements will be. This is just a starting point, and I tried to make it as simple as possible.

Reply:
I am with Lukas here, I would rather have these directly on the toolbar build method than to introduce an anti pattern like that, even if it is an initial version.

Also, I bet that allowing external sources to add custom tool bar items isn't something we need to care about in this initial version

Reply:
I don't think that we should use patterns that aren't recommended whatever the future intention is though.
For a simple starting point these widgets could just be put into the build method, since the current method isn't a feasible for modifications anyways? And then we can work out later how it should be structured for allowing users to add their own buttons here?

Reply:
Ok, I removed those static lists.